pom.xml

```diff
@@ -53,7 +53,7 @@
     <jenkins-core.version>${jenkins.version}</jenkins-core.version>
     <jenkins-war.version>${jenkins.version}</jenkins-war.version>
     <jenkins-test-harness.version>2.22</jenkins-test-harness.version>
-    <hpi-plugin.version>1.122</hpi-plugin.version>
+    <hpi-plugin.version>2.0</hpi-plugin.version>
```

Comment:
2.0 seems to be compiled for java 1.8 thus plugins fail to be compiled on 1.7

Reply:
Is there a reason you're using JDK 7 to build? AFAICT we've required JDK 8 to build for quite a while now (even when building for JRE 7).

Reply:
I think it's hard to test JRE 7 compatibility if compile maven step fails. I just submitted a pull request to the logstash plugin and it has CI system that tests with jdk7 still. So I couldn't use version `3.2` but `2.9`. Do you have an official doc that I can show to plugin owner, where it is explained building plugins for jre7 should be done with JDK 8?

Reply:
> we've required JDK 8 to build for quite a while now

And made explicit in #89.

> it's hard to test JRE 7 compatibility

If you mean running functional tests on JRE 7, that is not supported by the harness. Tests must be run on 8 and it is assumed that the plugin behaves similarly on 7 (which is EOL anyway and not recommended even for older Jenkins versions which support it). Setting `java.level=7` assures only that

* the bytecode format is readable by JRE 7
* no calls are made to APIs introduced in Java Platform 8 unless marked `@IgnoreJRERequirement`

> it has CI system that tests with jdk7 still

Patch its `.travis.yml` then. Note that as of https://github.com/jenkins-infra/pipeline-library/pull/2, plugins using the organization’s recommended `Jenkinsfile` will normally build on JDK 8 only.